logical/src/main/java/org/apache/drill/common/expression/fn/JodaDateValidator.java

```diff
@@ -0,0 +1,216 @@
+  public static final String JODA_FULL_NAME_OF_DAY = "EEEE";
+  public static final String JODA_DAY_OF_YEAR = "D";
+  public static final String JODA_DAY_OF_MONTH = "d";
+  public static final String JODA_DAY_OF_WEEK = "e";
+  public static final String JODA_NAME_OF_MONTH = "MMMM";
+  public static final String JODA_ABR_NAME_OF_MONTH = "MMM";
+  public static final String JODA_FULL_ERA_NAME = "G";
+  public static final String JODA_NAME_OF_DAY = "E";
+  public static final String JODA_TIME_ZONE_NAME = "TZ";
+  public static final String JODA_HOUR_12_NAME = "h";
+  public static final String JODA_HOUR_12_OTHER_NAME = "h";
+  public static final String JODA_HOUR_24_NAME = "H";
+  public static final String JODA_MINUTE_OF_HOUR_NAME = "m";
+  public static final String JODA_SECOND_OF_MINUTE_NAME = "s";
+  public static final String JODA_MILLISECOND_OF_MINUTE_NAME = "SSS";
+  public static final String JODA_WEEK_OF_YEAR = "w";
+  public static final String JODA_MONTH = "MM";
+  public static final String JODA_HALFDAY = "aa";
+
+  static {
+    postgresToJodaMap.put(POSTGRES_FULL_NAME_OF_DAY, JODA_FULL_NAME_OF_DAY);
+    postgresToJodaMap.put(POSTGRES_DAY_OF_YEAR, JODA_DAY_OF_YEAR);
+    postgresToJodaMap.put(POSTGRES_DAY_OF_MONTH, JODA_DAY_OF_MONTH);
+    postgresToJodaMap.put(POSTGRES_DAY_OF_WEEK, JODA_DAY_OF_WEEK);
+    postgresToJodaMap.put(POSTGRES_NAME_OF_MONTH, JODA_NAME_OF_MONTH);
+    postgresToJodaMap.put(POSTGRES_ABR_NAME_OF_MONTH, JODA_ABR_NAME_OF_MONTH);
+    postgresToJodaMap.put(POSTGRES_FULL_ERA_NAME, JODA_FULL_ERA_NAME);
+    postgresToJodaMap.put(POSTGRES_NAME_OF_DAY, JODA_NAME_OF_DAY);
+    postgresToJodaMap.put(POSTGRES_TIME_ZONE_NAME, JODA_TIME_ZONE_NAME);
+    postgresToJodaMap.put(POSTGRES_HOUR_12_NAME, JODA_HOUR_12_NAME);
+    postgresToJodaMap.put(POSTGRES_HOUR_12_OTHER_NAME, JODA_HOUR_12_OTHER_NAME);
+    postgresToJodaMap.put(POSTGRES_HOUR_24_NAME, JODA_HOUR_24_NAME);
+    postgresToJodaMap.put(POSTGRES_MINUTE_OF_HOUR_NAME, JODA_MINUTE_OF_HOUR_NAME);
+    postgresToJodaMap.put(POSTGRES_SECOND_OF_MINUTE_NAME, JODA_SECOND_OF_MINUTE_NAME);
+    postgresToJodaMap.put(POSTGRES_MILLISECOND_OF_MINUTE_NAME, JODA_MILLISECOND_OF_MINUTE_NAME);
+    postgresToJodaMap.put(POSTGRES_WEEK_OF_YEAR, JODA_WEEK_OF_YEAR);
+    postgresToJodaMap.put(POSTGRES_MONTH, JODA_MONTH);
+    postgresToJodaMap.put(POSTGRES_HALFDAY_AM, JODA_HALFDAY);
+    postgresToJodaMap.put(POSTGRES_HALFDAY_PM, JODA_HALFDAY);
+  }
+
+  static {
+    postgresValuesForDeleting.add(SUFFIX_SP);
+    postgresValuesForDeleting.add(PREFIX_FM);
+    postgresValuesForDeleting.add(PREFIX_FX);
+    postgresValuesForDeleting.add(PREFIX_TM);
+  }
+
+  /**
+   * Replaces all postgres patterns from {@param pattern},
+   * available in postgresToJodaMap keys to jodaTime equivalents.
+   *
+   * @param pattern date pattern in postgres format
+   * @return date pattern with replaced patterns in joda format
+   */
+  public static String toJodaFormat(String pattern) {
```

Comment:
Do we support escape characters? This is also necessary if the format allows escape characters. While Postres is a big vague on escapes, this is how it works for Oracle: http://stackoverflow.com/questions/588430/what-is-the-best-way-to-escape-non-format-characters-in-oracles-to-char

Escapes throw off simple pattern matching. This is a date format: "SS". This is not: ""SS"".